.github/workflows/ci.yml

```diff
@@ -132,16 +137,18 @@ jobs:
           name: docs
           path: docs/build
 
-
   # ----------------------------------------------------------------------------------------------------
   # --- Integration and Unit Testing -------------------------------------------------------------------
   # ----------------------------------------------------------------------------------------------------
 
-  Test-Suite:
+  Test-Suite-Trusted:
```

Comment:
Be default, for a PR from a fork, both will be in a skipped state which is treated as "success" for merge check by GitHub (in a case where we add both `Test-Sute-Trusted` and `Test-Sute-Fork` to the checks). This might cause aviator to merge even when we have not got the CI to pass.

There are two options that I can think to deal with this.

#### Option 1

We can have a single Test-Suite job with a step inside which will check if `!(github.event.pull_request.head.repo.full_name == github.repository || github.event_name == 'repository_dispatch')` and fail with an error message, something like "Need member approval to run this job.". This way, we can keep the current flow.

### Option 2

Have two jobs but have another job which just runs and returns true if one of these is a success. This way we can target that job to the mergeability check. One caveat is that we will have to make both of the Test-Suite branches fail/succeed (not skip) so that the dependent job can run.

Reply:
>  This might cause aviator to merge even when we have not got the CI to pass.

We shouldn't be auto-merging (aviator) PRs from forks right? 🤔 

Even if all tests pass...

Reply:
Yeah, I guess we shouldn't be auto merging. Now that you mention it, I can see an issue where we add the `mergequeue` label, somebody can push a new commit, on top of which aviator might merge master and it will cause it to run making the non-verified code run. That said, I think having this check could be helpful in even preventing us from accidentally merging it in.

Reply:
I agree. I can update the rule on the main branch to require the `Test-Suite-Fork` be required? So if it's in a skipped state and cannot be merged (can only do this after it is merged into `main`).

@ntolia can you confirm if aviators merge rules prevent it from auto-merging PRs from forks. I do not have access.

Reply:
@uncledru Aviator does not seem to differentiate between PRs from branches/vs forks. It just uses the GitHub merability check to decide if a PR should go in or not.

Reply:
Do you have access to `aviator`? Maybe we can add [merge rules](https://docs.aviator.co/reference/merge-rules) to prevent PRs from forks being auto-merged?

Reply:
Aviator is currently set to use the same rules as what GitHub uses to check for mergeability.
![2022-11-16-13-03-59](https://user-images.githubusercontent.com/14259816/202116021-ac3df986-fee7-4a81-a430-a1214b4adb9a.png)

Reply:
If you click "customize" does it give any options that could help us out here?

Reply:
We can specify which jobs we want to check by adding more jobs to block on. As of now this list is synced with what is checked in GitHub. I don't have the access to GitHub settings but if you can update the list, it will update them here as well.
![2022-11-18-09-13-33](https://user-images.githubusercontent.com/14259816/202611988-0bdeb0cf-e526-48e7-a73a-cbe41f8d81dc.png)

Reply:
Done: 

<img width="753" alt="image" src="https://user-images.githubusercontent.com/7441475/202727087-6fdacf4c-efc7-4cd1-84ee-18b6ce53bca1.png">

Reply:
I will remove the `Test-Suite` once this is merged.

Reply:
@meain I had to revert the required checks since it was stopping corso PRs from being mergable. After this is merged (approval still needed) - we can configure the repo and aviator to respect these new checks.

Reply:
The changes looks good. My concern here was weather we should consider making `Test-Sute-Fork` fail when it has not been approved so as to not accidentally merge it. As of not it is skipped when the PR is ran without approval and GitHub treats skipped and successful test in the same way.

Reply:
🤔 if we make it a required check in GitHub Settings - it should be prevent the PR being merged even if its skipped. 

After the `ok-to-test` comment is added - Actions should update the check. 

Do you have a working example of what you're talking about? (allowing the PR to merge even if a required check is skipped)

Reply:
Never mind... https://docs.github.com/en/actions/using-jobs/using-conditions-to-control-job-execution

<img width="799" alt="image" src="https://user-images.githubusercontent.com/7441475/203070080-fda0c78c-73ea-4eef-b9de-fc2d9456b94b.png">

Any ideas/suggestions on making Test-Sute-Fork fail when it has not been approved?

Reply:
I'm thinking we can remove the `if` condition on the `Test-Suite-Fork` job and have it always run. 

The first step in the job could be a bash script that runs the same logic checks as the current `if`:

```yaml
    if: github.event_name == 'repository_dispatch' &&
      github.event.client_payload.slash_command.sha != '' &&
      contains(github.event.client_payload.pull_request.head.sha, github.event.client_payload.slash_command.sha)
```

If this resolves to `false`... We exit the job with `exit 1` and it'll get registered as a failure which will prevent merging. 

```bash
exit 1
```

After a developer comments `ok-to-test` it will run again but this time via `workflow_dispatch`... the logical check above will evaluate to `true` and we won't exit early. 

Thoughts?

Reply:
Yup! That was more or less what I had in mind as well.